### PR TITLE
feat: introduce `talos.experimental.wipe` kernel param to wipe system disk

### PR DESCRIPTION
**Pull request by Unix4ever**
Fixes: https://github.com/talos-systems/talos/issues/4399

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4734)

### CODE REVIEW

**Review by smira — COMMENTED** [review]
we need docs for this (we have a document which lists all available kernel args) and a note in `release.toml`.

**Comment by andrewrynhard** [comment]
what is `exp`?

**Comment by Unix4ever** [comment]
> what is exp?

`experimental` I guess

**Comment by Unix4ever** [comment]
> note in `release.toml`.

Ok, then I need [this](https://github.com/talos-systems/talos/pull/4735) to be merged :)

**Comment by andrewrynhard** [comment]
> > what is exp?
> 
> `experimental` I guess

I know `exp` is less to type, but I would say we should use `experimental`. It describes itself and in general this is something I think we should do/are doing in Talos (using the full names and not shortening them). Just my two cents. This is a minor nit.

**Comment by Unix4ever** [comment]
> > > what is exp?
> > 
> > `experimental` I guess
> 
> I know `exp` is less to type, but I would say we should use `experimental`. It describes itself and in general this is something I think we should do/are doing in Talos (using the full names and not shortening them). Just my two cents. This is a minor nit.

Sure, why not. Looks like the limit is 4096 bytes, so maybe we shouldn't be stingy about that limit.

File: website/content/docs/v0.15/Guides/resetting-a-machine.md [hunk 1]
```diff
@@ -25,3 +27,11 @@ The `graceful` flag is especially important when considering HA vs. non-HA Talos
 If the machine is part of an HA cluster, a normal, graceful reset should work just fine right out of the box as long as the cluster is in a good state.
 However, if this is a single node cluster being used for testing purposes, a graceful reset is not an option since Etcd cannot be "left" if there is only a single member.
 In this case, reset should be used with `--graceful=false` to skip performing checks that would normally block the reset.
+
+## Kernel Parameter
```
Comment (smira) [inline website/content/docs/v0.15/Guides/resetting-a-machine.md:31]: :+1:

File: website/content/docs/v0.15/Guides/resetting-a-machine.md [hunk 2]
```diff
@@ -25,3 +27,11 @@ The `graceful` flag is especially important when considering HA vs. non-HA Talos
...
+
+Another way to reset a machine is to specify `talos.experimental.wipe=system` kernel parameter.
+If the machine got stuck in the boot loop and you access to the console you can use GRUB to specify this kernel argument.
+Then when Talos boots for the next time it will reset system disk and reboot.
+
+Next steps can be to install Talos either usign PXE boot or by mounting an ISO.
```
Comment (smira) [inline website/content/docs/v0.15/Guides/resetting-a-machine.md:37]: could you please also add it to https://github.com/talos-systems/talos/blob/05fd16adfde59550b3f8982262319250871390f7/website/content/docs/v0.15/Reference/kernel.md

Comment (lion7) [reply website/content/docs/v0.15/Guides/resetting-a-machine.md:37]: There is a typo in the last sentence: usign instead of using 😉

**Review by smira — APPROVED** [review]

**Pull request merged**